tests/inverted-index-test.js

```diff
@@ -51,6 +221,105 @@ describe('Search index', () => {
   });
 
   it('should return [-1] for any word not present', () => {
-    expect(indexObj.searchIndex(indexObj.index, 'book1.json', 'catwalk')).toEqual({ 'catwalk': [-1] });
+    expect(indexObj.searchIndex(indexObj.index, 'book1.json', 'catwalk')).toEqual({ catwalk: [-1] });
+  });
+});
+
+
+describe('create index endpoint', () => {
+  it('should return an error message for malformed file', () => {
+    supertest(app)
+    .post('/api/create')
+    .field('fileName', 'malformedBook.json')
+    .attach('fileContent', path.join('fixtures', 'malformedBook.json'))
+    .expect({ error: 'Index could not be created, uploaded file must be a valid JSON file and file name must have .json extension' })
+        .end((err, res) => {
+          if (err) {
+            throw err;
+          }
+        });
+  });
+
+  it('should create correct index for any file uploaded', () => {
+    supertest(app)
+    .post('/api/create')
+    .field('fileName', 'book1.json')
+    .attach('fileContent', path.join('fixtures', 'book1.json'))
+    .expect(createdIndexForBook1)
+        .end((err, res) => {
+          if (err) {
+            throw err;
+          }
+        });
+  });
+
+  it('should return "kindly upload a file" when no file is uploaded', () => {
+    supertest(app)
+    .post('/api/create')
+    .field('fileName', 'book1.json')
+    .expect({ error: 'Kindly upload a file' })
+        .end((err, res) => {
+          if (err) {
+            throw err;
+          }
+        });
+  });
+
+  it('should return "Invalid file uploaded" when no file is uploaded', () => {
+    supertest(app)
+    .post('/api/create')
+    .field('fileName', 'invalidBook.txt')
+    .attach('fileContent', path.join('fixtures', 'invalidBook.txt'))
+    .expect({ error: 'Invalid file uploaded!' })
+        .end((err, res) => {
+          if (err) {
+            throw err;
+          }
+        });
+  });
+});
+
+
+describe('search index endpoint', () => {
+  it('should return "Index has not been created for the specified file" if index has not been created', (done) => {
+    supertest(app)
+    .post('/api/search')
+    .send({ fileName: 'book3.json', searchTerms: 'the' })
+    .expect({ error: 'Index has not been created for the specified file' })
+        .end((err, res) => {
+          if (err) {
+            done.fail(err);
+          } else {
+            done();
+          }
+        });
+  });
+
+  it('should return correct result if both fileName and searchTerms are supplied', (done) => {
+    supertest(app)
+    .post('/api/search')
+    .send({ fileName: 'book1.json', searchTerms: 'best' })
+    .expect({ best: [0] })
+        .end((err, res) => {
+          if (err) {
+            done.fail(err);
+          } else {
+            done();
+          }
+        });
+  });
+
```

Comment:
Can't find test that "Ensure searchIndex can handle a varied number of search terms as arguments"